### PR TITLE
Remove impossible events

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/impl/event_factory.py [hunk 1]
```diff
@@ -571,13 +571,7 @@ def deserialize_message_reaction_remove_all_event(
...
             )
 
-        # TODO: check if this can even occur.
-        return reaction_events.DMReactionDeleteAllEvent(
-            app=self._app,
-            shard=shard,
-            channel_id=channel_id,
-            message_id=message_id,
-        )
+        raise NotImplementedError("DM reaction all removes are not documented behavior")
```
Comment (FasterSpeeding) [inline hikari/impl/event_factory.py:574]: Yes it is?

**Comment by FasterSpeeding** [comment]
This just breaks forward compatability

File: hikari/impl/event_factory.py (hunk 1 above)
Comment (davfsa) [reply hikari/impl/event_factory.py:574]: I haven't been able to trigger any of the two events 🤔

Comment (FasterSpeeding) [reply hikari/impl/event_factory.py:574]: They are documented behaviour though

Comment (davfsa) [reply hikari/impl/event_factory.py:574]: Hmmmm. We could just keep them ig

**Pull request closed without merging**